src/lexer/mod.rs

```diff
@@ -230,7 +230,12 @@ impl Token {
                     let token_type = if is_numeric(&ident_or_number) {
                         TokenKind::BigInt(ident_or_number)
                     } else if is_hexadecimal(&ident_or_number) {
-                        TokenKind::Hex(ident_or_number)
+                        let hex = ident_or_number.trim_start_matches("0x");
+                        TokenKind::BigInt(
+                            BigInt::from_str_radix(hex, 16)
+                                .expect("invalid hex number")
```

Comment:
I think we should return a proper error here

Reply:
It shouldn't be possible to get an error here since `is_hexadecimal` would return false. Maybe return error if the number starts with `0x` and rest is not a valid hexadecimal value?